### PR TITLE
Add guidelines for authorship of the MOOSE paper 

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #26320

@idaholab/moose-ccb

### CODE REVIEW

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+MOOSE is a collaborative effort and not only do we always welcome contributions, but also want to
+credit them appropriately and in a sustainable manner. We do this by including our contributors
+in the relevant journal article(s) that should be used by all, both within the community and externally,
+to cite the MOOSE framework and its modules. Please see the [Citing.md] page for the current version
+of these articles. These articles are published on a regular basis.
+
+## MOOSE framework paper authorship guidelines
+
+When citing MOOSE, whether when using it, using a derived application, or just referring to
+the MOOSE project, this is the article that should be cited. It is featured in the [Citing.md] page.
```
Comment (cticenhour) [inline modules/doc/content/framework_development/authorship_guidelines.md:12]:
```suggestion
the MOOSE project, the top-level MOOSE paper is the article that should be cited. It is featured in the [Citing MOOSE](Citing.md#MOOSE) page.
```

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+The contributors are listed in two categories.
+
+- Semi-permanent contributors are included in every new MOOSE framework paper by default. These contributors
+  have written a very large part of the MOOSE framework and must be credited, at least until their contributions
+  are overtaken by others.
+
+The current criteria for semi-permanent authorship are listed below:
+
+- Top 10 line of codes in the `framework/` and other utility folders or
```
Comment (cticenhour) [inline modules/doc/content/framework_development/authorship_guidelines.md:29]:
```suggestion
- Top 10 contributor as measured by lines of code in the `framework/` and other utility folders or
```

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+- Top 10 commits in the `framework/` folder
```
Comment (cticenhour) [inline modules/doc/content/framework_development/authorship_guidelines.md:30]:
```suggestion
- Top 10 contributor as measured by number of commits in the `framework/` folder
```

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+
+!alert note
+While it seems poor practice to recognize contribution by LOC or commit numbers, this criteria did not
```
Comment (cticenhour) [inline modules/doc/content/framework_development/authorship_guidelines.md:34]:
```suggestion
While it seems poor practice to recognize contribution by lines of code or commit numbers, this criteria did not
```

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+exist when these contributions were made, so the metric did not influence the coding style of these authors.
+These criteria may be reformulated in the future in terms of features.
```
Comment (cticenhour) [inline modules/doc/content/framework_development/authorship_guidelines.md:36]:
```suggestion
These criteria may be reformulated in the future in terms of contributed features and impact.
```

Just a suggestion.

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+
+The current guidelines for transient authorship are listed below:
+
+- 1 major contribution or
+- 4 minor contributions and significant involvement in MOOSE-related activities or
+- 10 fixup contributions and significant involvement in MOOSE-related activities
+
+
+Contributions are only tallied for the *framework and supporting utilities*. Modules are deliberately excluded
```
Comment (cticenhour) [inline modules/doc/content/framework_development/authorship_guidelines.md:50]:
```suggestion
Contributions are only tallied for the *framework and supporting utilities*. Modules are deliberately excluded,
```

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+as developers of a module should publish a module paper and report on new developments outside of the MOOSE framework paper.
+MOOSE-related activities are developments in the MOOSE modules and MOOSE applications.
+
+A list of major, minor and fixup example contributions has been developed as guidelines for
+classification and can be shared upon request. The classification of contributions is decided when
+gathering the list of authors for the next paper.
+
+!alert note
+Contributors who do not meet the criteria for authorship in the paper will see their contributions from
+the past period considered for the next paper.
```
Comment (cticenhour) [inline modules/doc/content/framework_development/authorship_guidelines.md:60]: The wording here is a bit unclear. Does this mean that a contributor who doesn't meet the guidelines for the new update paper will have their contributions from the same period considered for the next paper (on top of new contributions in the intervening period)?

Consider revising.

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+we may consider them at the level of minor and fixup contributions.
+
+!alert note
+The contribution guidelines may be scaled depending on the time elapsed between successive publications. The current
+guidelines were developed for an 18 months time frame between two publications.
+
+## MOOSE Modules paper authorship guidelines
+
+Modules should strive to follow the same principles as the framework with regards to authorship inclusivity
+and representativity. However, most modules are not developed by the framework team and have their own policies.
```
Comment (cticenhour) [inline modules/doc/content/framework_development/authorship_guidelines.md:73]: I would add something about contacting the member of the CCB related to a given module (with a link to the CODEOWNERS file perhaps?)

**Review by cticenhour — DISMISSED** [review]
Thanks for writing this up! A few suggestions and comments below.

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 7 above)
Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md:60]: yes that would be my intention. 2 over 1 period then 2 over the next makes 4 for the next

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 8 above)
Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md:73]: Codeowner s file s pretty rough git no?

Comment (cticenhour) [reply modules/doc/content/framework_development/authorship_guidelines.md:73]: A list of module directories next to GitHub handles? I don't think that's hard to parse for someone who is a contributor. Adding a link is just a suggestion, though.

Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md:73]: I dont think it matches each module's CCB that well. Reactor module for example, I would say most of their CCB is at ANL and not a codeowner because git still isnt developing this obvious need of non-maintainer codeowners

File: framework/doc/content/framework/contributing.md [hunk 9]
```diff
@@ -13,6 +13,10 @@ engaging in the community will certainly be beneficial.
 
 !alert-end!
 
+!alert title=Authorship in MOOSE-development paper
```
Comment (cticenhour) [inline framework/doc/content/framework/contributing.md:16]:
```suggestion
!alert note title=Authorship in MOOSE-development paper
```

This should fix up the errors on CI

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
+# MOOSE papers authorship guidelines
+
+MOOSE is a collaborative effort and not only do we always welcome contributions, but also want to
+credit them appropriately and in a sustainable manner. We do this by including our contributors
+in the relevant journal article(s) that should be used by all, both within the community and externally,
+to cite the MOOSE framework and its modules. Please see the [Citing.md] page for the current versions
```
Comment (GiudGiud) [inline modules/doc/content/framework_development/authorship_guidelines.md:6]:
```suggestion
to cite the MOOSE framework and its modules. Please see the [citing.md] page for the current versions
```

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+of these articles. These articles are published and updated on a regular basis.
+
+## MOOSE framework paper authorship guidelines
+
+When citing MOOSE, whether when using it, using a derived application, or just referring to
+the MOOSE project, the top-level MOOSE paper is the article that should be cited. It is featured in the [Citing MOOSE](Citing.md#MOOSE) page.
```
Comment (GiudGiud) [inline modules/doc/content/framework_development/authorship_guidelines.md:12]:
```suggestion
the MOOSE project, the top-level MOOSE paper is the article that should be cited. It is featured in the [Citing MOOSE](citing.md#MOOSE) page.
```

File: framework/doc/content/framework/contributing.md [hunk 12]
```diff
@@ -13,6 +13,10 @@ engaging in the community will certainly be beneficial.
 
 !alert-end!
 
+!alert note title=Authorship in MOOSE-development paper
```
Comment (lindsayad) [inline framework/doc/content/framework/contributing.md:16]:
```suggestion
!alert note title=Authorship in MOOSE-development papers
```

Comment (lindsayad) [reply framework/doc/content/framework/contributing.md:16]: To help denote that there are multiple kinds of MOOSE papers, including a framework paper, module papers, etc.

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+MOOSE is a collaborative effort and not only do we always welcome contributions, but also want to
+credit them appropriately and in a sustainable manner. We do this by including our contributors
+in the relevant journal article(s) that should be used by all, both within the community and externally,
+to cite the MOOSE framework and its modules. Please see the [citing.md] page for the current versions
+of these articles. These articles are published and updated on a regular basis.
+
+## MOOSE framework paper authorship guidelines
+
+When citing MOOSE, whether when using it, using a derived application, or just referring to
+the MOOSE project, the top-level MOOSE paper is the article that should be cited. It is featured in the [Citing MOOSE](citing.md#MOOSE) page.
```
Comment (lindsayad) [inline modules/doc/content/framework_development/authorship_guidelines.md:12]:
```suggestion
the MOOSE project, the top-level MOOSE framework paper should be cited. It is featured in the [Citing MOOSE](citing.md#MOOSE) page.
```

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+This paper is regularly replaced with either a new paper or an "update" paper. "Update" papers
+refer to a summary of changes related to a periodic update of the software from the timeframe of the previous "update" paper, and can have a different authorship list as the original
```
Comment (lindsayad) [inline modules/doc/content/framework_development/authorship_guidelines.md:14]:
```suggestion
refer to a summary of changes related to a periodic update of the software from the timeframe of the previous "update" paper, and can have a different authorship list than the original
```

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+The current criteria for semi-permanent authorship are listed below:
+
+- Top 10 contributor as measured by lines of code in the `framework/` and other utility folders or
+- Top 10 contributor as measured by number of commits in the `framework/` folder
+
+
+!alert note
+While it seems poor practice to recognize contribution by lines of code or commit numbers, this criteria did not
+exist when these contributions were made, so the metric did not influence the coding style of these authors.
+These criteria may be reformulated in the future in terms of contributed features and impact.
```
Comment (lindsayad) [inline modules/doc/content/framework_development/authorship_guidelines.md:36]: I don't think this kind of note is necessary. If you want a note here, I might say that pull request reviewers will ask that duplicate/extraneous code be removed or refactored and that meaningless commits be squashed.

Comment (lindsayad) [reply modules/doc/content/framework_development/authorship_guidelines.md:36]: Or if you want to keep some of the content here that's ok, but I would consider mixing in my suggestions, and noting that we have always valued clean code and clean commit history (even if those values haven't 100% been honored all the time)

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+The current guidelines for transient authorship are listed below:
+
+- 1 major contribution or
+- 4 minor contributions and significant involvement in MOOSE-related activities or
+- 10 fixup contributions and significant involvement in MOOSE-related activities
+
+
+Contributions are only tallied for the *framework and supporting utilities*. Modules are deliberately excluded,
+as developers of a module should publish a module paper and report on new developments outside of the MOOSE framework paper.
+MOOSE-related activities are developments in the MOOSE modules and MOOSE applications.
```
Comment (lindsayad) [inline modules/doc/content/framework_development/authorship_guidelines.md:52]: I think this sentence is confusing on the backs of the other two. It could be less confusing if you edit the bulleted list to say "1 major framework contribution", "4 minor framework contributions", etc.

Comment (lindsayad) [reply modules/doc/content/framework_development/authorship_guidelines.md:52]: If you do that, then I would delete the first sentence of this paragraph and then probably swap what are currently sentences two and three

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+
+!alert note
+The contribution guidelines may be scaled depending on the time elapsed between successive publications. The current
+guidelines were developed for an 18 months time frame between two publications.
+
+## MOOSE Modules paper authorship guidelines
+
+Modules should strive to follow the same principles as the framework with regards to authorship inclusivity
+and representativity. However, most modules are not developed by the framework team and have their own policies.
+Please contact the relevant module team for more information.
```
Comment (lindsayad) [inline modules/doc/content/framework_development/authorship_guidelines.md:74]:
```suggestion
Please contact the relevant module code owners for more information.
```

Comment (lindsayad) [reply modules/doc/content/framework_development/authorship_guidelines.md:74]: I think this makes it more tangible who one should contact

**Review by lindsayad — COMMENTED** [review]
thanks for taking the time to write this!

File: modules/doc/content/framework_development/authorship_guidelines.md [hunk 18]
```diff
@@ -0,0 +1,74 @@
+# MOOSE papers authorship guidelines
+
+MOOSE is a collaborative effort and not only do we always welcome contributions, but also want to
+credit them appropriately and in a sustainable manner. We do this by including our contributors
+in the relevant journal article(s) that should be used by all, both within the community and externally,
+to cite the MOOSE framework and its modules. Please see the [citing.md] page for the current versions
```
Comment (GiudGiud) [inline modules/doc/content/framework_development/authorship_guidelines.md:6]: the "and its modules" part might be problematic. We want modules to be cited individually. 

Prefer
- cite the MOOSE framework.
- cite MOOSE
?

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 17 above)
Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md:74]: I ll add a note. I dont consider code owners to be close enough to the team for each module

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 15 above)
Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md:36]: re-worded. Maybe not everyone has to know that historically we did not use LOC / commits for authorship (though I suspect we did lol)

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 17 above)
Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md]: see now

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 18 above)
Comment (lindsayad) [reply modules/doc/content/framework_development/authorship_guidelines.md:6]: What about "cite the MOOSE framework and any modules used" ?

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 17 above)
Comment (lindsayad) [reply modules/doc/content/framework_development/authorship_guidelines.md]: > I dont consider code owners to be close enough to the team for each module

Lol, what?! Then we don't have the correct code owners

Comment (lindsayad) [reply modules/doc/content/framework_development/authorship_guidelines.md]: I would hope that the code owners listed are good representatives of the module teams. If you're saying that not all module team members are listed as code owners, then I would agree with that.

Comment (lindsayad) [reply modules/doc/content/framework_development/authorship_guidelines.md]: I like what you wrote, so I'm resolving this convo

Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md]: Reactor team at ANL has no code owner.

Fluid properties is similarly a not full match of people in charge. 

I m sure tensor mechanics and phase field similarly have 1/2 the people we actually consider to be decision makers for those efforts in the codeowners file

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 18 above)
Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md:6]: I get what you mean that using a module you should still cite moose framework 

But citing a moose module externally from a publication that did not use it, you probably don't need to cite moose framework (though it does not hurt)

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 17 above)
Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md]: Ah i started typing too early. Yes you got what I mean

File: modules/doc/content/framework_development/authorship_guidelines.md (hunk 18 above)
Comment (GiudGiud) [reply modules/doc/content/framework_development/authorship_guidelines.md:6]: oh nevermind this is a paragraph on both moose and its modules. this works

**Comment by GiudGiud** [comment]
Making sure we are all in agreement here.

**Review by lindsayad — DISMISSED** [review]

File: framework/doc/content/framework/contributing.md [hunk 19]
```diff
@@ -13,6 +13,10 @@ engaging in the community will certainly be beneficial.
 
 !alert-end!
 
+!alert note title=Authorship in MOOSE-development papers
+MOOSE being a collaborative effort, the authorship of MOOSE is shared within the community. Please see the [authorship_guidelines.md](modules/doc/content/framework_development/authorship_guidelines.md optional=True)
```
Comment (hugary1995) [inline framework/doc/content/framework/contributing.md:17]: This link doesn't work for me in the PR documentation website.

Comment (GiudGiud) [inline framework/doc/content/framework/contributing.md:17]:
```suggestion
MOOSE being a collaborative effort, the authorship of MOOSE is shared within the community. Please see the [authorship_guidelines.md](framework_development/authorship_guidelines.md optional=True)
```

**Review by hugary1995 — APPROVED** [review]

File: framework/doc/content/framework/contributing.md [hunk 20]
```diff
@@ -13,6 +13,10 @@ engaging in the community will certainly be beneficial.
 
 !alert-end!
 
+!alert note title=Authorship in MOOSE-development papers
+MOOSE being a collaborative effort, the authorship of MOOSE is shared within the community. Please see the [authorship_guidelines.md](framework_development/authorship_guidelines.md optional=True)
```
Comment (GiudGiud) [inline framework/doc/content/framework/contributing.md:17]: if it still doesnt work I might just hard code. This would actually make it work for all downstream apps ;)

Comment (hugary1995) [inline framework/doc/content/framework/contributing.md:17]:
```suggestion
MOOSE being a collaborative effort, the authorship of MOOSE is shared within the community. Please see the [authorship guidelines](framework_development/authorship_guidelines.md optional=True)
```

Comment (GiudGiud) [reply framework/doc/content/framework/contributing.md]: works now! thanks for checking @hugary1995

**Comment by GiudGiud** [comment]
I ll wait until more people have seen this and agreed. With the holidays it might be ten days before people look at it though

**Review by lindsayad — APPROVED** [review]

**Review by roystgnr — APPROVED** [review]
Thanks for the wait

**Comment by GiudGiud** [comment]
last chance to comment, merging Tuesday otherwise
